core/deployment/src/main/java/io/quarkus/deployment/configuration/BuildTimeConfigurationReader.java

```diff
@@ -1117,6 +1121,41 @@ public Set<String> getPropertyNames() {
             return properties;
         }
 
+        /**
+         * Use this Config instance to record the runtime default values. We cannot use the main Config
+         * instance because it may record values coming from local development sources (Environment Variables,
+         * System Properties, etc.) in at build time. Local config source values may be completely different between the
```

Comment:
can we be a  bit more specific here than just "etc" - what are we actually using from recording?

looking at code below we are not excluding sources but just gathering  a list of sources...what is actually defining what we include/exclude here?

Reply:
Yes, we build a new Config instance, excluding specific sources. The `.env` sources are also of type `EnvConfigSource`, so they get excluded (or, to be more correct, not added to the config instance used for recording).